src/precice/Participant.hpp

```diff
@@ -22,27 +22,180 @@ struct WhiteboxAccessor;
 
 namespace precice {
 
-/// convenience type for compatibility
+/** @brief forwards-compatible typedef for C++17 \ref std::string_view.
+ *
+ * The \ref string_view is a read-only view into sequence of characters.
+ *
+ * A good mental model for this type is a struct containing a pointer and a size like the following:
+ *
+ * @code{cpp}
+ * struct string_view {
+ *   const char* first;
+ *   size_t size;
+ * };
+ * @endcode
+ *
+ * It can be constructed from:
+ * - a pointer to a NULL-terminated C-string
+ * - a pointer and a size
+ * - a pointer to the first and one past the last element
+ * - \ref std::string can directly convert itself to \ref std::string_view.
+ *
+ * In practise, using a function such as \ref Participant::getMeshDimensions() that expects a string_view looks like this:
+ *
+ * You can directly pass the string literal:
+ *
+ * @code{.cpp}
+ * getMeshDimensions("MyMesh");
+ * @endcode
+ *
+ * Or you can pass something string-like:
+ *
+ * @code{.cpp}
+ * // one of
+ * const char*      meshName = "MyMesh";
+ * std::string      meshName = "MyMesh";
+ * std::string_view meshName = "MyMesh";
+ * // followed by
+ * getMeshDimensions(meshName);
```

Comment:
I'm not sure we should encourage people to use `string_view` or `char*` in their code. While all variants here work, I think the `std::string` (or the literals) are the way to go in CPP, as the user code typically also owns the data.

Reply:
> The point is to show that string_view can also be passed to the preCICE API.

Should be clear from the function signature, no? What you describe here is nice to know, but actually expected user knowledge in case they care about it I would say.